Add venue details page

nikomakela opened the pull request:
LIIKUNTA-336. Migrated the old Liikunta-Helsinki venue details page components to fit in their new context inside the monorepo. Tried to use some common components and not to copy components that would duplicate any features.

NOTE: Testing this PR before the #109 and #110 are deployed needs a local Apollo Router, because otherwise the Venue-API won't be available!

Add the next line in the `.env.local`

```NEXT_PUBLIC_FEDERATION_ROUTER_ENDPOINT="http://localhost:4000"```

and run the Apollo Router locally:

```yarn docker:graphql-router:sports:serve --build```

<img width="671" alt="image" src="https://user-images.githubusercontent.com/389204/207971739-ef92dfb3-2224-4c2e-b132-73897fb10eea.png">

<img width="667" alt="image" src="https://user-images.githubusercontent.com/389204/207971784-7c0b8ec0-546e-4831-adfc-677049fe0b05.png">


----
Multi-step deployment, because the current CI environment cannot handle a concurrent deployment for router and app changes:

1. Copy and migrate the Apollo-Server for Venues and Hauki data from the old deprecated Liikunte-Helsinki repo: https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/110
2. Do the Apollo Router work for the added Venue-subgraph (added in step 1): https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/109
3. Add (copy and migrate) the Venue details page when the Venue datasource is available (added in step 2): https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/108

nikomakela committed:
feat(sports): add the subgraph support

LIIKUNTA-336

nikomakela committed:
chore(sports): venue details page

LIIKUNTA-336

nikomakela committed:
feat(sports): service map component for events and venues

Migrate the service map implementation into a shape
that the same code can be used for the venues and the events

nikomakela committed:
feat(sports): added venue translations

LIIKUNTA-336

nikomakela committed:
refactor(sports): event details page translations and infoblocks

LIIKUNTA-336. Took venue details page translations in use.
Refactored the venue details page components and
removed (almost) all the imports of infoblock-component -
used the same components as in event details page instead.

nikomakela committed:
refactor(sports): return path handling of the return button

LIIKUNTA-336.
Refactored and fixed the return path handling and the return button control
in event and venue details pages.

nikomakela committed:
test: fix the failing tests after the huge venue migration

LIIKUNTA-336

nikomakela commented on apps/sports-helsinki/src/domain/graphql/createApolloServer.ts:
```diff
@@ -20,15 +22,15 @@ const isProd = process.env.NODE_ENV === 'production';
-    _empty: String
-  }
-`;
+// const initQueryTypeDefs = gql`
+//   type Query {
+//     _empty: String
+//   }
+// `;
```
I guess it has been a test whether it's needed still or not. Seems like not!

tuomas777 reviewed (approved):
![clap](https://user-images.githubusercontent.com/1314604/208638264-9d0d9439-7515-4893-a54d-fab0c116b9dc.gif)

nikomakela committed:
fix: integrity and code quality issues

LIIKUNTA-336
fix: typecheck issues
fix: linter issues
chore(sports): remove unnecessary comment lines

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr108.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr108.dev.hel.ninja** :rocket::rocket::rocket:
